mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -999,18 +1000,18 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
     }
 
     private fun showMissingPermissionsWarningIfNeeded() {
-        val tasksWithPermissions = mutableMapOf(PrefKeys.SEND_PHONE_STATE to Manifest.permission.READ_PHONE_STATE)
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            tasksWithPermissions[PrefKeys.SEND_WIFI_SSID] = Manifest.permission.ACCESS_FINE_LOCATION
-        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            tasksWithPermissions[PrefKeys.SEND_WIFI_SSID] = Manifest.permission.ACCESS_COARSE_LOCATION
-        }
+        val tasksWithPermissions = listOf(
+            PrefKeys.SEND_PHONE_STATE,
+            PrefKeys.SEND_WIFI_SSID
+        )
 
         val missingPermissions = tasksWithPermissions
             .filter { entry ->
-                prefs.getStringOrNull(entry.key)?.toItemUpdatePrefValue()?.first == true && !hasPermission(entry.value)
+                val requiredPermissions = BackgroundTasksManager.getRequiredPermissionsForTask(entry)
+                prefs.getStringOrNull(entry)?.toItemUpdatePrefValue()?.first == true &&
+                    requiredPermissions != null && !hasPermissions(requiredPermissions)
             }
-            .map { entry -> entry.value }
+            .map { entry -> BackgroundTasksManager.getRequiredPermissionsForTask(entry) }
```

Comment:
@maniac103 Do you have a hint for me on how to convert this list of string arrays in an array?

Reply:
Didn't actually try, but I think there's `.flatten().toSet()`.

Reply:
`.toTypedArray().flatten().toSet()` should do it, but I haven't tested it yet. Thanks for the hint.